LabTools/CoreWidgets/RemoteEnableWidget.py

```diff
@@ -0,0 +1,386 @@
+        if SCROLLABLE:
+            # create scrollarea for stacked:
+            self.scrollarea = QtGui.QScrollArea()
+            self.scrollLayout = QtGui.QVBoxLayout()
+            self.scrollLayout.setSpacing(0)
+            self.scrollLayout.setContentsMargins(0,0,0,0)
+            for state in self.REN_Lines:
+                self.scrollLayout.addWidget(state)
+            self.scrollwidget = QtGui.QWidget()
+            self.scrollwidget.setLayout(self.scrollLayout)
+            self.scrollarea.setWidget(self.scrollwidget)
+            self.verticalLayout.addWidget(self.scrollarea)  # instead of stacked
+        else:
+            # self.verticalLayout.addStretch()
+            for state in self.REN_Lines:
+                self.verticalLayout.addWidget(state)
+            # self.verticalLayout.addStretch()
+ #       #self.footer = self.create_footer()
+##        self.verticalLayout.addLayout(self.footer)
+
+        self.setLayout(self.verticalLayout)
+
+    def send_mode(self, mode):
+        instr = self.current_instrument()
+
+        if instr is not None:
+            instr.set_ren(mode)
+
+    def set_default(self, mode, from_widget = False):
+        if self.default_ren != mode and self.default_ren is not None:
+            self.REN_Lines[self.default_ren].default_state(False)
+        self.default_ren = mode
+        self.default_widget.set_default(mode)
+        # TODO: add stuff to change default text
+        if not from_widget:
+            for i in range(len(REN_MODES)):
+                if self.default_ren != i:
+                    self.REN_Lines[i].default_state(False)
+                else:
+                    self.REN_Lines[i].default_state(True)
+        else:
+            self.save_default()
+
+    def unset_default(self, mode, from_widget = False):
+        if self.default_ren == mode:
+            self.default_ren = None
+            self.default_widget.set_default(None)
+            # TODO: add stuff to change default text
+        if not from_widget:
+            for i in range(len(REN_MODES)):
+                self.REN_Lines[i].default_state(False)
+        else:
+            self.save_default()
+
+    def unset_all(self, from_widget = False):
+        self.default_ren = None
+        for state in self.REN_Lines:
+            state.default_state(False)
+
+        self.default_widget.set_default(None)
+
+        if from_widget:
+            self.save_default()
+
+    def save_default(self):
+        if self.parentClass is not None:
+            self.parentClass.save_ren()
+
+
+    # following two functions are used from LabGui.py to load/save defaults
+    def get_ren(self):
+        return self.default_ren
+
+    def set_ren(self, level):
+        if level is None or level == 'None':
+            self.default_ren = None
+            self.unset_all()
+        else:
+            self.set_default(int(level)) # this will automatically update default_ren, as well as unsetting all
+
```

Comment:
unsetting all what?

Reply:
The way this widget works is that there can be a default REN enabled for all machines, which will automatically assert for each one on connection. When a user selects a level as the default, set_default(level) will automatically set the default REN value to the specified level for all devices, as well as graphically represent the change within the widget.

If the user wishes to remove a default REN, by passing level=None, it will remove the default (by setting it equal to None) and graphically represent the change within the widget by stating there is no default, by disabling the "Clear" button, and by updating each "line" such that the corresponding state has the option to be set as the default.

Reply:
This is a good description of what REN is useful for ! You could reuse it in the docstring of the widget :)

My comment was targeted to the sentence of the comment `# this will automatically update default_ren, as well as unsetting all`, did you mean `all level` then?